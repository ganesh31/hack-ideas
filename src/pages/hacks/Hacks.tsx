import { useEffect, useState } from "react";
import invariant from "tiny-invariant";
import { getAllHacks } from "../../api/hack/hack";
import { getAllTags } from "../../api/tag/tag";
import Card from "../../components/card/Card";
import { Hack } from "../../types/hack";
import { Tag } from "../../types/tag";

interface Props {}

const Hacks: React.FC<Props> = (props: Props) => {
  const [allHacks, setAllHacks] = useState<Hack[]>([]);
  const [allTags, setAllTags] = useState<Tag[]>([]);

  useEffect(() => {
    async function fetch() {
      const hacks = await getAllHacks();
      invariant(hacks, "no hacks available");
      setAllHacks(hacks);

      const tags = await getAllTags();
      invariant(tags, "no tags available");
      setAllTags(tags);
    }
    fetch();
  }, []);

  return (
    <div className=" flex flex-col items-center">
      <section className="grid-cols-auto-fit grid w-full gap-2">
        {allHacks.length > 0 ? (
          allHacks.map((hack) => {
            return (
              <Card
                data-testid="hack"
                key={hack.id}
                allTags={allTags}
                hack={hack}
                onToggleLike={() => {}}
              />
            );
          })
        ) : (
          <>
            <h1 className="text-center font-semibold ">
              Sorry no hacks are currently available.
            </h1>
            <p className="italic text-center">
              Waiting for someone to add some hacks
            </p>
          </>
        )}
      </section>
    </div>
  );
};

export default Hacks;
