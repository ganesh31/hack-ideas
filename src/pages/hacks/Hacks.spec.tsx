import { render, screen, waitFor } from "@testing-library/react";
import Hacks from "./Hacks";

describe("Hacks", () => {
  it("should show no hacks if there are no hacks", async () => {
    const { asFragment } = await render(<Hacks />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render all the hacks", async () => {
    const { asFragment } = render(<Hacks />);

    await waitFor(() => {
      expect(screen.getByText(/Jan Circuits '22/i)).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
