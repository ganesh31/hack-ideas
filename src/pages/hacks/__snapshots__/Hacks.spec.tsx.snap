// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hacks should render all the hacks 1`] = `
<DocumentFragment>
  <div
    class=" flex flex-col items-center"
  >
    <section
      class="grid-cols-auto-fit grid w-full gap-2"
    >
      <div
        class="group border-slate-800 hover:bg-slate-200 bg-slate-50 text-slate-800 hover:cursor-pointer min-w-[350px] sm:min-w-sm p-4 space-y-1 border rounded shadow"
      >
        <div
          class="flex justify-between"
        >
          <div
            class="text-lg font-bold"
          >
            Jan Circuits '22
          </div>
          <div
            class="flex space-x-1"
          >
            <span
              class=""
            >
              1
            </span>
            <svg
              aria-hidden="true"
              class="text-red-600 w-5 h-5"
              data-testid="button"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 10h4.764a2 2 0 011.789 2.894l-3.5 7A2 2 0 0115.263 21h-4.017c-.163 0-.326-.02-.485-.06L7 20m7-10V5a2 2 0 00-2-2h-.095c-.5 0-.905.405-.905.905 0 .714-.211 1.412-.608 2.006L7 11v9m7-10h-2M7 20H5a2 2 0 01-2-2v-6a2 2 0 012-2h2.5"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </div>
        <p
          class="text-ellipsis whitespace-nowrap overflow-hidden"
        >
          Solve a challenge and win exiting prizes ðŸŽ‰
        </p>
        <div
          class="group-hover:opacity-100 flex justify-between text-sm transition duration-300 ease-in-out opacity-0"
        >
          <span>
            <i>
              Sat Feb 12 2022
            </i>
          </span>
        </div>
        <div
          class="space-x-1 overflow-hidden"
        />
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Hacks should show no hacks if there are no hacks 1`] = `
<DocumentFragment>
  <div
    class=" flex flex-col items-center"
  >
    <section
      class="grid-cols-auto-fit grid w-full gap-2"
    >
      <h1
        class="text-center font-semibold "
      >
        Sorry no hacks are currently available.
      </h1>
      <p
        class="italic text-center"
      >
        Waiting for someone to add some hacks
      </p>
    </section>
  </div>
</DocumentFragment>
`;
